Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,254 @@
+
+            return new VcamData
+            {
+                id = id,
+                vcam_class = GetVcamTypeName(vcam),
+                has_follow_target = vcam.Follow != null,
+                has_lookat_target = vcam.LookAt != null,
+                blend_hint = vcam.m_Transitions.m_BlendHint.ToString(),
+                inherit_position = vcam.m_Transitions.m_InheritPosition,
+                standby_update = vcam.m_StandbyUpdate.ToString(),
+                mode_overwrite = vcam.m_Lens.ModeOverride.ToString(),
+                body_component = bodyComponent,
+                aim_component = aimComponent,
+                noise_component = noiseComponent,
+                custom_component_count = customComponentCount,
+                extensions = vcamExtensions.ToArray(),
+                custom_extension_count = customExtensionCount,
+            };
+        }
+
+        static void GetExtensions(CinemachineVirtualCameraBase vcamBase, 
+            out List<string> vcamExtensions, out int customCount)
+        {
+            customCount = 0;
+
+            // collect extensions on vcam
+            vcamExtensions = new List<string>();
+            var extensions = vcamBase.mExtensions;
+            if (extensions != null)
+            {
+                for (var i = 0; i < extensions.Count; i++)
+                {
+                    vcamExtensions.Add(GetTypeName(extensions[i].GetType(), ref customCount));
+                }
+            }
+        }
+
+        static void GetComponents(CinemachineVirtualCamera vcam,
+            out string body, out string aim, out string noise, out int customCount)
+        {
+            customCount = 0;
+            body = aim = noise = "";
+            var cmComps = vcam.GetComponentPipeline();
+            if (cmComps != null)
+            {
+                for (var i = 0; i < cmComps.Length; i++)
+                {
+                    var componentName = GetTypeName(cmComps[i].GetType(), ref customCount);
+                    switch (cmComps[i].Stage)
+                    {
+                        case CinemachineCore.Stage.Body:
+                            body = componentName;
+                            break;
+                        case CinemachineCore.Stage.Aim:
+                            aim = componentName;
+                            break;
+                        case CinemachineCore.Stage.Noise:
+                            noise = componentName;
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+        
+        const string k_CustomStr = "Custom"; // used for hiding user created class names
+        static string GetVcamTypeName(CinemachineVirtualCameraBase vcamBase)
+        {
+            var type = vcamBase.GetType();
+            return s_CinemachineAssembly == type.Assembly ? type.Name : k_CustomStr;
+        }
```

Comment:
You don't need this method, or s_CinemachineAssembly, or even k_CustomStr

Reply:
Why?

Reply:
Because it's unnecessarily repeated logic.  That should be painful to you.  You can call GetTypeName(vcamType, out var _);

Reply:
static string GetTypeName(Type type, ref int customTypeCount)

I cannot pass _ for customTypeCount, so I would need to write another function:
static string GetTypeName(Type type)

Reply:
I could also do the following as an overload to not have repeated logic:
```
static string GetTypeName(Type type)
        {
            int temp = 0;
            return GetTypeName(type, ref temp);
        }
```